Sources/IRGen/Expression/IRSubscriptExpression.swift

```diff
@@ -43,33 +45,34 @@ struct IRSubscriptExpression {
 
     switch subExpr.baseExpression {
     case .identifier:
-      return runtimeFunc(String(baseOffset), indexExpressionCode)
+      return (preamble, runtimeFunc(String(baseOffset), indexExpressionCode.expression))
     case .subscriptExpression(let newBase):
-      return runtimeFunc(nestedStorageOffset(subExpr: newBase,
-                                             baseOffset: baseOffset,
-                                             functionContext: functionContext),
-                         indexExpressionCode)
+      let e = nestedStorageOffset(subExpr: newBase,
```

Comment:
If this returned `GeneratedCode` then you wouldn't have to reference tuple members by index further down, which would make this far more readable